packages/remix-react/components.tsx

```diff
@@ -1309,3 +1310,18 @@ function useComposedRefs<RefValueType = any>(
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, refs);
 }
+
+let OutletContext = React.createContext<AppData>(null);
+
+export function Outlet<Data = AppData>({ context }: { context?: Data }) {
+  return (
+    <OutletContext.Provider value={context}>
+      <RouterOutlet />
+    </OutletContext.Provider>
+  );
+}
+
+export function useOutletContext<ContextData>() {
```

Comment:
```suggestion
export function useOutletData<ContextData>() {
```
useOutletData?

Reply:
I think I prefer sticking to the termonology of `context` rather than `data` because I'm really concerned that people will unnecessarily use this in place of what's arguably a better solution for accessing the parent's loader data (via `useMatches`).